packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -233,6 +243,46 @@ export interface RecordSetOptions {
   readonly setIdentifier?: string;
 }
 
+/**
+ * Configuration for IP-based routing.
+ */
+export interface ICidrRoutingConfig {
+  /**
+   * List of CIDR blocks.
+   *
+   * When specifying the default location (locationName is '*'), it cannot be set, but for all other cases, it is mandatory.
+   *
+   * @default - zero bit CIDR block (0.0.0.0/0 or ::/0) for the default location
+   */
+  cidrList?: string[];
+  /**
+   * The name of the location.
+   *
+   * When '*' is specified, it is treated as the default location.
+   * In this case, the cidrList cannot be specified.
+   */
+  locationName: string;
```

Comment:
We should't have two props that are reliant on the value of the other to work or not work. Since these two seem to be tightly coupled, we want the contract to enforce that.